crates/bytecode/src/iterator.rs

```diff
@@ -0,0 +1,173 @@
+use crate::{opcode, Bytecode, OpCode};
+
+/// Iterator over opcodes in a bytecode, skipping immediates.
+///
+/// This allows you to iterate through the actual opcodes in the bytecode,
+/// without dealing with the immediate values that follow PUSH instructions.
+#[derive(Debug, Clone)]
+pub struct BytecodeIterator<'a> {
+    /// Reference to the underlying bytecode bytes
+    bytes: &'a [u8],
+    /// Current position in the bytecode
+    position: usize,
+    /// End position in the bytecode (to handle original length for legacy bytecode)
+    end: usize,
+}
+
+impl<'a> BytecodeIterator<'a> {
+    /// Creates a new iterator from a bytecode reference.
+    pub fn new(bytecode: &'a Bytecode) -> Self {
+        let bytes = bytecode.bytecode();
+        let end = match bytecode {
+            Bytecode::LegacyAnalyzed(analyzed) => analyzed.original_len(),
```

Comment:
Bytecode::Eip7702 variant is different as it contains an address. We could put its end to be `0` so no iteration would happen.